framework/src/neml2/userobjects/NEML2Assembly.C

```diff
@@ -0,0 +1,117 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef NEML2_ENABLED
+
+// Torch includes
+#include <ATen/ops/from_blob.h>
+
+// MOOSE includes
+#include "NEML2Assembly.h"
+
+using namespace libMesh;
+
+registerMooseObject("MooseApp", NEML2Assembly);
+
+InputParameters
+NEML2Assembly::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
+
+  params.addClassDescription(
+      "This user object gathers the JxWxT values from all elements in the assembly and "
+      "provides them as a neml2 tensor. This is useful for assembling NEML2 models that "
+      "require the JxWxT values for each element.");
+
+  ExecFlagEnum execute_options = MooseUtils::getDefaultExecFlagEnum();
+  execute_options = {EXEC_INITIAL, EXEC_LINEAR};
```

Comment:
we need linear instead of nonlinear/something less often because you are assuming displacements are variables?

Reply:
Primarily because this is at this time only for explicit time integration!